src/client/lib/hooks/useRecaptcha.tsx

```diff
@@ -173,6 +173,43 @@ const useRecaptcha: UseRecaptcha = (
 
 export default useRecaptcha;
 
+/**
+ * Wrapper component for the useRecaptcha hook.
+ *
+ * Runs the hook in a wrapper and passes the state back up to the parent component.
+ * Useful for instances where we want to conditionally include the reCAPTCHA check.
+ *
+ * @returns The reCAPTCHA form element.
+ */
+export const RecaptchaWrapper: React.FC<{
+  recaptchaSiteKey: string;
+  setRecaptchaState: React.Dispatch<
+    React.SetStateAction<UseRecaptchaReturnValue | undefined>
+  >;
+}> = ({ recaptchaSiteKey, setRecaptchaState }) => {
+  const { error, executeCaptcha, expired, requestCount, token, widgetId } =
+    useRecaptcha(recaptchaSiteKey, 'recaptcha');
+  React.useEffect(() => {
+    setRecaptchaState({
+      error,
+      executeCaptcha,
+      expired,
+      requestCount,
+      token,
+      widgetId,
+    });
+  }, [
+    error,
+    executeCaptcha,
+    expired,
+    requestCount,
+    setRecaptchaState,
+    token,
+    widgetId,
```

Comment:
what's the advantage of using 'useEffect' when everything is a dependency?

Reply:
good point, I can't see any reason why it's needed here

Reply:
Ah that's a good reason, calling setRecaptchaState is probably triggering the rerender above this